tls/client-tls-cryptocb.c

```diff
@@ -427,108 +104,156 @@ int main(int argc, char** argv)
+
+    /* validate peer certificate */
+    wolfSSL_CTX_set_verify(ctx, WOLFSSL_VERIFY_PEER, NULL);
 
     /* register a devID for crypto callbacks */
     wolfSSL_CTX_SetDevId(ctx, devId);
 
+#if 0
+    /* Example: "TLS13-AES256-GCM-SHA384", "TLS13-AES128-GCM-SHA256" or "TLS13-CHACHA20-POLY1305-SHA256" */
+    wolfSSL_CTX_set_cipher_list(ctx, "TLS13-AES256-GCM-SHA384");
+#endif
+
     /* Load client certificates into WOLFSSL_CTX */
     if ((ret = wolfSSL_CTX_load_verify_locations(ctx, CA_FILE, NULL))
-         != SSL_SUCCESS) {
+         != WOLFSSL_SUCCESS) {
         fprintf(stderr, "ERROR: failed to load %s, please check the file.\n",
                 CA_FILE);
-        goto ctx_cleanup;
+        goto exit;
     }
 
     /* Create a WOLFSSL object */
     if ((ssl = wolfSSL_new(ctx)) == NULL) {
         fprintf(stderr, "ERROR: failed to create WOLFSSL object\n");
         ret = -1;
-        goto ctx_cleanup;
+        goto exit;
     }
 
     /* Attach wolfSSL to the socket */
     if ((ret = wolfSSL_set_fd(ssl, sockfd)) != WOLFSSL_SUCCESS) {
         fprintf(stderr, "ERROR: Failed to set the file descriptor\n");
-        goto cleanup;
+        goto exit;
     }
 
     /* Connect to wolfSSL on the server side */
-    if ((ret = wolfSSL_connect(ssl)) != SSL_SUCCESS) {
+    if ((ret = wolfSSL_connect(ssl)) != WOLFSSL_SUCCESS) {
         fprintf(stderr, "ERROR: failed to connect to wolfSSL\n");
-        goto cleanup;
+        goto exit;
     }
 
+    cipher = wolfSSL_get_current_cipher(ssl);
+    printf("SSL cipher suite is %s\n", wolfSSL_CIPHER_get_name(cipher));
+
     /* Get a message for the server from stdin */
     printf("Message for server: ");
     memset(buff, 0, sizeof(buff));
     if (fgets(buff, sizeof(buff), stdin) == NULL) {
         fprintf(stderr, "ERROR: failed to get message for server\n");
         ret = -1;
-        goto cleanup;
+        goto exit;
     }
     len = strnlen(buff, sizeof(buff));
 
     /* Send the message to the server */
     if ((ret = wolfSSL_write(ssl, buff, len)) != len) {
         fprintf(stderr, "ERROR: failed to write entire message\n");
         fprintf(stderr, "%d bytes of %d bytes were sent", ret, (int) len);
-        goto cleanup;
+        goto exit;
     }
 
     /* Read the server data into our buff array */
     memset(buff, 0, sizeof(buff));
     if ((ret = wolfSSL_read(ssl, buff, sizeof(buff)-1)) == -1) {
         fprintf(stderr, "ERROR: failed to read\n");
-        goto cleanup;
+        goto exit;
     }
 
     /* Print to stdout any data the server sends */
     printf("Server: %s\n", buff);
 
-    ret = 0;
+    ret = 0; /* return success */
 
+exit:
     /* Cleanup and return */
-cleanup:
-    wolfSSL_free(ssl);      /* Free the wolfSSL object                  */
-ctx_cleanup:
-    wolfSSL_CTX_free(ctx);  /* Free the wolfSSL context object          */
+    if (sockfd != SOCKET_INVALID)
+        close(sockfd);          /* Close the connection to the server       */
+    if (ssl != NULL)
```

Comment:
Don't need to check pointer, wolfSSL_free() should of the check
Same with wolfSSL_CTX_free().

Reply:
Fixed